src/webgl/light.js

```diff
@@ -494,6 +494,33 @@ p5.prototype.pointLight = function(v1, v2, v3, x, y, z) {
   return this;
 };
 
+p5.prototype.imageLight = function(img){
+  // ! if the graphic is always true, do we really need this section?
```

Comment:
If we're going to use `img` as a light source, it means we'll need the blurry version of its texture in order to pass into the shader. If `this.blurryTextures(img)` is defined, then we've already made the blurry texture in a previous frame and don't need to do anything more here. If not though, that's when we need to run the commented out code below to make it, and then after line 512 we'll need to store it with `this.blurryTextures.set(img, newGraphic)`

Reply:
We could alternatively do this if statement right before we do `shader.setUniform('equiRectangularTextures', this.blurryTextures.get(this.activeImageLight));` later on. The idea is that we need that texture to exist by that point